src/nvim/ui_compositor.c

```diff
@@ -716,6 +716,11 @@ static void ui_comp_event(UI *ui, char *name, Array args)
       handled = true;
     }
   })
+  if (err.type != kErrorTypeNone) {
```

Comment:
we should do this inside of the loop. multiple `event_cb->cb` could error out. also `api_clear_error()` should not be in an else branch.

Reply:
@bfredl, thanks for reviewing the PR.

I saw the iteration issue when I first made the change, and tried the below:

```C
map_foreach_value(&ui_event_cbs, event_cb, {
  Error err = ERROR_INIT;
  Object res = nlua_call_ref(event_cb->cb, name, args, false, &err);
  if (res.type == kObjectTypeBoolean && res.data.boolean == true) {
    handled = true;
  }
  if (err.type != kErrorTypeNone) {
    ELOG("Error while executing ui_comp_event callback: %s", err.msg);
  }
  api_clear_error(&err);
})
```

but I was getting the compiler error of `too many arguments provided to function-like macro invocation`. I'm a C novice and wasn't sure how to approach it. With your review, I've made an attempt at handling the error for each iteration. Please let me know if there's a better way to do it.

Thank you for your time, I appreciate it.